### PR TITLE
smi metrics istio installation through helm

### PR DESCRIPTION
**Pull request by Pothulapati**
This PR 
- Adds one more option for Istio i.e `default` which dosen't add any more metrics during the installation and goes with the default Istio metrics (considering that they have added the req labels). This is useful when users dont want to add more metrics.
- For SMI-Metrics to just work out of the box with Istio, they can leave the default flag and it will install the required metrics and use them to answer the queries. No changes to Istio installation required in this case. 
- Moves Istio.yaml and Linkerd.yaml to the `templates/` directory as they are added based on the `adapter` option.

Signed-off-by: Tarun Pothulapati <tarunpothulapati@outlook.com>

### CODE REVIEW

File: chart/values.yaml [hunk 1]
```diff
@@ -5,6 +5,5 @@ image:
   pullPolicy: IfNotPresent
```
Comment (grampelberg) [inline chart/values.yaml]: Can we get comments in this file pointing to the provider values? I'd be confused as to all the missing defaults here.

File: docs/istio.md [hunk 2]
```diff
@@ -2,7 +2,20 @@
...
+## Out of the Box Installation
+
+As Istio allows metrics to be configurable, SMI with Istio can be installed by directly running
+
+```bash
+helm template chart --set adapter=istio | k apply -f -
+```
+This installs the necessary instances, handlers and rules for Istio to emit those metrics along with SMI-Metrics APIServer.
+
+## Using the default Istio Metrics
```
Comment (grampelberg) [inline docs/istio.md]: Should probably note that the chart requires `istio` is already installed and running.

File: chart/values.yaml (hunk 1 above)
Comment (Pothulapati) [reply chart/values.yaml]: Done

File: docs/istio.md (hunk 2 above)
Comment (Pothulapati) [reply docs/istio.md]: Noted

**Pull request merged**